### ISSUE TITLE
Converting a script into a module

### ISSUE THREAD

**Issue by wadudmiah** [TACC/Lmod#598]
Hi,

I am using Lmod 8.7.7 and am trying to convert a script into a module file using the command which complains with an error:
```
$ /usr/share/lmod/lmod/libexec/sh_to_modulefile /local/software/OpenFOAM/7.0/OpenFOAM-7/etc/bashrc > 7.0.lua
Error in script "/local/software/OpenFOAM/7.0/OpenFOAM-7/etc/bashrc":

$
```
However, I am able to source the script with no issues:
```
$ source /local/software/OpenFOAM/7.0/OpenFOAM-7/etc/bashrc
$
``` 
Is there a bug in this version of Lmod?

**Comment by rtmclay** [comment]
Lmod requires that the script pass "set -e".  This means that every statement must have a status of 0, (i.e. no error).  Try running in a new shell:
```
$ set -exv ;  source /local/software/OpenFOAM/7.0/OpenFOAM-7/etc/bashrc
```
and see where it fails and fix it.
In OpenFoam 10 I found the following error as I reported on the Lmod mailing list:

-----------------------------------------------------------------------------

I have found the issue is with the OpenFOAM script. Here is the current problem:  Currently there are these lines in etc/config.sh/aliases starting at line 74:
```
if command -V wmRefresh 2> /dev/null | head -1 | grep -q "function"
then
    unset wmRefresh
else
    unalias wmRefresh 2> /dev/null
fi
```
When there is no alias for wmRefresh the if test returns a false because there is nothing written to stdout.  So the script tries to run "unalias wmRefresh" which fails and sets the error flag.  Because the shell conversion program does "set -e", means that the shell script conversion stops at the unalias line. But there is no message, because the error message is written to /dev/null.

I could remove the "set -e" when evaluating the script but then I won't know if the conversion fails and I'm unable to report that error.  The fix here is to change the script to:
```
if command -V wmRefresh 2> /dev/null | head -1 | grep -q "function"
then
    unset wmRefresh
else
    unalias wmRefresh 2> /dev/null || true
fi
```
The error message in the latest version of Lmod is better than what was in Lmod 8.7.7
